Src/DependencyCollector/Shared/ServiceBusDiagnosticsListener.cs

```diff
@@ -119,10 +112,12 @@ private void SetCommonProperties(string eventName, object eventPayload, Activity
         {
             // activity name looks like 'Microsoft.Azure.ServiceBus.<Name>'
             // as namespace is too verbose, we'll just take the last node from the activity name as telemetry name
-            string[] activityNameSegments = activity.OperationName.Split('.');
-            if (activityNameSegments.Length > 0)
+            string activityName = activity.OperationName;
+            int lastDotIndex = activityName.LastIndexOf('.');
+            if (lastDotIndex > 0)
             {
-                telemetry.Name = activityNameSegments[activityNameSegments.Length - 1];
+                int length = activityName.Length - lastDotIndex - 1;
```

Comment:
You don't need to compute length yourself if you use the single-arg overload of Substring (you just give it the start position and it goes to the end of the string)